src/main/java/eu/europeana/metis/sandbox/service/workflow/HarvestServiceImpl.java

```diff
@@ -46,6 +55,41 @@
     return records;
   }
 
+  @Override
+  public List<ByteArrayInputStream> harvest(String endpoint, String setSpec, String prefix) throws ServiceException {
+
+    List<ByteArrayInputStream> records = new ArrayList<>();
+    List<Pair<String, Exception>> exceptions = new ArrayList<>();
+
+    try {
+      OaiRecordHeaderIterator recordHeaderIterator = harvesterOai
+          .harvestRecordHeaders(new OaiHarvest(endpoint, prefix, setSpec));
+      OaiRepository oaiRepo = new OaiRepository(endpoint, prefix);
+
+      recordHeaderIterator.forEach(rh -> {
```

Comment:
I do not see any record number limitation, this can stay here for a very long time..

Reply:
Maybe this was the issue you were having before Adolfo?

Reply:
For this there is a new ticket to handle records limitations better with a message to the user.